src/worker.ts

```diff
@@ -1,70 +1,83 @@
         });
       }
+    }
 
-      const webhookPayload = await request.json();
-      const settings = Value.Decode(pluginSettingsSchema, Value.Default(pluginSettingsSchema, webhookPayload.settings));
+    if (request.method !== "POST") {
+      return new Response(JSON.stringify({ error: `Only POST requests are supported.` }), {
+        status: 405,
+        headers: { "content-type": "application/json", Allow: "POST" },
+      });
+    }
+    const contentType = request.headers.get("content-type");
+    if (contentType !== "application/json") {
+      return new Response(JSON.stringify({ error: `Error: ${contentType} is not a valid content type` }), {
+        status: 400,
+        headers: { "content-type": "application/json" },
+      });
+    }
+    let payload;
 
-      if (!pluginSettingsValidator.test(settings)) {
-        const errors: string[] = [];
-        for (const error of pluginSettingsValidator.errors(settings)) {
-          console.error(error);
-          errors.push(`${error.path}: ${error.message}`);
-        }
-        return new Response(JSON.stringify({ error: `Error: "Invalid settings provided. ${errors.join("; ")}"` }), {
-          status: 400,
-          headers: { "content-type": "application/json" },
-        });
+    try {
+      payload = await request.clone().json();
+    } catch (err) {
+      return new Response(JSON.stringify({ error: "Invalid JSON payload", err }), {
+        status: 400,
+        headers: { "content-type": "application/json" },
+      });
+    }
+    const envSettings = Value.Decode(envValidator.schema, Value.Default(envValidator.schema, env));
+    if (!envValidator.test(envSettings)) {
+      const errors: string[] = [];
+      for (const error of envValidator.errors(envSettings)) {
+        console.error(error);
+        errors.push(`${error.path}: ${error.message}`);
       }
-      if (!envValidator.test(env)) {
-        const errors: string[] = [];
-        for (const error of envValidator.errors(env)) {
-          console.error(error);
-          errors.push(`${error.path}: ${error.message}`);
-        }
-        return new Response(JSON.stringify({ error: `Error: "Invalid environment provided. ${errors.join("; ")}"` }), {
+      return new Response(JSON.stringify({ error: `Error: "Invalid environment provided. ${errors.join("; ")}"` }), {
+        status: 400,
+        headers: { "content-type": "application/json" },
+      });
+    }
+
+    // inits the worker with the telegram bot
+    await TelegramBotSingleton.initialize(envSettings);
+
+    try {
+      if (isGithubPayload(payload)) {
+        // inits the worker with the plugin context for this call
+        PluginContext.initialize(payload, envSettings);
+        await handleGithubWebhook(request, envSettings);
+      } else if (isTelegramPayload(payload)) {
+        await handleTelegramWebhook(request, envSettings);
+      } else {
+        return new Response(JSON.stringify({ error: "Invalid environment provided" }), {
```

Comment:
instead of accepting both telegram and kernel requests on the same path and trying to figure out which one is it, why not have a separate path like `/webhook/telegram` and `/webhook/kernel`?

Reply:
Wouldn't that require that the kernel change the path it's targeting? 

`/webhook` works for catching telegram hooks tho. I've been forwarding payloads locally to `localhost:3000/webhook` using smee, maybe I have got my wires crossed and I'm doing something wrong, e.g I should have some sort of check here like `path === /webhook => handleTgWebhook` instead of `isTgPayload => handleTgWebhook`?

https://github.com/ubiquibot/telegram-bot/blob/4f690b3cdb74673673ed3a81b769c38d8d58e2d8/src/server/index.ts#L62-L65

Reply:
in the case of worker plugins you specify the URL in the config so it can have a custom path.

i'm a bit confused now, why do we need a hono server and also a worker server?

Reply:
> in the case of worker plugins you specify the URL in the config so it can have a custom path.

Are you suggesting that in the kernel config we do `.../webhook/github` and we set the webhook URL on the telegram bot as `.../webhook/telegram`?

> i'm a bit confused now, why do we need a hono server and also a worker server?

Simply put, Hono isn't a server itself it handles routing and middleware mostly built with CF in mind. The reason why I split them like this was because 1. Kernel plugins have no framework but we have a sort of format for them. 2. The TG frameworks used support various frameworks to run your bot on and I've been wanting to use Hono in a plugin for a while as it has good reviews.

https://chatgpt.com/share/66ebe5b8-19e0-8000-a20a-2af9bb6292bc

I am open to suggestions on different approaches such as whether to remove the Hono dependency

Reply:
> Are you suggesting that in the kernel config we do `.../webhook/github` and we set the webhook URL on the telegram bot as `.../webhook/telegram`?

Yes



> Simply put, Hono isn't a server itself it handles routing and middleware mostly built with CF in mind. The reason why I split them like this was because 1. Kernel plugins have no framework but we have a sort of format for them. 2. The TG frameworks used support various frameworks to run your bot on and I've been wanting to use Hono in a plugin for a while as it has good reviews.

Actually the [kernel SDK](https://github.com/ubiquity/ubiquibot-kernel/blob/development/src/sdk/server.ts) uses Hono, although I haven't implemented the option to change the path, it can be implemented quickly. 

If you want to use Hono with Cloudflare workers you can just make a [default export](https://hono.dev/docs/getting-started/cloudflare-workers#_2-hello-world) of the Hono app and it should work out of the box

Reply:
> Actually the [kernel SDK](https://github.com/ubiquity/ubiquibot-kernel/blob/development/src/sdk/server.ts) uses Hono

Oh nice I didn't know that. 

> If you want to use Hono with Cloudflare workers you can just make a [default export](https://hono.dev/docs/getting-started/cloudflare-workers#_2-hello-world) of the Hono app and it should work out of the box

I did know this but as I said I wanted to keep the typical plugin structure and only use Hono for the bot routes/middleware which isn't the main entry but I'm happy to refactor the plugin side and use Hono for everything. It'll be another nice little experiment in this repo 😂

Reply:
Please I will be super grateful if my telegram account is been unbanned through this platform. The telegram number is as follows; +2348066388388. Thank you all